logs-to-tsv/write-logs-tsv.js

```diff
@@ -0,0 +1,70 @@
+const { readFileSync, writeFileSync, readdirSync } = require('fs')
+
+const convertToTSV = (arr) => {
+  const array = [Object.keys(arr[0])].concat(arr)
+  return array.map((log) => {
+    return Object.values(log).join('\t')
+  }).join('\n')
+}
+
+function returnMatch (log, match) {
+  if (log.match(match)) {
+    return log.match(match)[1]
+  } else return null
+}
+
+const regExpressions = {
+  timestamp: /\[(.*?)\]/,
+  searchScope: /&search_scope=(.*?)$/,
+  query: /q=+(.*?)(&|$)/,
+  contributor: /contributor=(.*?)(&|$)/,
+  title: /title=(.*?)(&|$)/,
+  subject: /subject=(.*?)(&|$)/
+}
+
+const parseLogs = (logs) => {
+  // loop through array of logs
+  return logs.map((log) => {
+    let request = decodeURIComponent(log)
+    return Object.keys(regExpressions).reduce((obj, param) => {
+      const extractedValue = returnMatch(request, regExpressions[param])
+      return {
+        ...obj, [param]: extractedValue
+      }
+    }, {})
+  })
+    .filter((request) => request)
+}
+
+const writeLogs = () => {
+  const logs = readdirSync('logs-to-tsv/logs-out')
+  const parsedLogs = logs.map((logFilePath) => {
+    const logFile = readFileSync('logs-to-tsv/logs-out/' + logFilePath, 'utf-8')
+    return parseLogs(logFile.split('\t'))
+  })
+  writeFileSync('logs-to-tsv/logsOut.tsv', convertToTSV(parsedLogs.flat()))
```

Comment:
Can we `console.log` something to say it successfully wrote to _?

Reply:
Note the path to the final log file also needs to drop the `logs-to-tsv` prefix. 

Also, picky, but everything else in this PR uses shish-kebab. Can the final result be `logs-out.tsv`?